pkg/splunk/client/azureblobclient.go

```diff
@@ -108,126 +207,242 @@ func NewAzureBlobClient(ctx context.Context, bucketName string, storageAccountNa
 		Prefix:             prefix,
 		StartAfter:         startAfter,
 		Endpoint:           endpoint,
-		Client:             splunkClient,
+		HTTPClient:         azureHTTPClient.(SplunkHTTPClient),
 	}, nil
 }
 
 // InitAzureBlobClientWrapper is a wrapper around InitAzureBlobClientSession
 func InitAzureBlobClientWrapper(ctx context.Context, appAzureBlobEndPoint string, storageAccountName string, secretAccessKey string) interface{} {
-	return InitAzureBlobClientSession(ctx, appAzureBlobEndPoint, storageAccountName, secretAccessKey)
+	return InitAzureBlobClientSession()
 }
 
 // InitAzureBlobClientSession initializes and returns a client session object
-func InitAzureBlobClientSession(ctx context.Context, appAzureBlobEndPoint string, storageAccountName string, secretAccessKey string) SplunkAzureBlobClient {
+func InitAzureBlobClientSession() SplunkHTTPClient {
+	return &http.Client{}
+}
+
+// Update http request header with secrets info
+func updateAzureHTTPRequestHeaderWithSecrets(ctx context.Context, client *AzureBlobClient, httpRequest *http.Request) error {
 	reqLogger := log.FromContext(ctx)
-	scopedLog := reqLogger.WithName("InitAzureBlobClientSession")
+	scopedLog := reqLogger.WithName("updateHttpRequestHeaderWithSecrets")
+
+	scopedLog.Info("Updating Azure Http Request with secrets")
 
-	scopedLog.Info("Initializing  AzureBlob Service for apps", "appAzureBlobEndPoint", appAzureBlobEndPoint)
-	var azureBlobClient SplunkAzureBlobClientTemp
+	// Update httpRequest header with data and version
+	httpRequest.Header[headerXmsDate] = []string{time.Now().UTC().Format(http.TimeFormat)}
+	httpRequest.Header[headerXmsVersion] = []string{azureHTTPHeaderXmsVersion}
+
+	// Get HMAC signature using storage account name and secret access key
+	stringToSign, err := buildStringToSign(*httpRequest, client.StorageAccountName)
+	if err != nil {
+		scopedLog.Error(err, "Azure Blob with secrets Failed to build string to sign")
+		return err
```

Comment:
nit: if the caller is handling the error, we can just return full error.

Reply:
This error would help us identify what exact part of the auth is failing. The error signature produced by the buildStringToSign might not imply the same.